### PR TITLE
ci: Added common CI check for server and client formatting

### PR DESCRIPTION
**Pull request by nidhi-nair**
Combined both server and client side checks into one workflow, conditionally triggered based on path of files changed in the PR. This is so we can have one compulsory check for all kinds of formatting in the repo.

### CODE REVIEW

File: .github/workflows/client-prettier.yml [hunk 1]
```diff
@@ -1,66 +1,33 @@
-name: Client Prettier
+name: Appsmith Client Prettier Check Workflow
```
Comment (sharat87) [inline .github/workflows/client-prettier.yml:1]: Can we please not prefix everything with `Appsmith`? It's already bad enough. 😬

shot-2023-07-20-13-07-19

File: .github/workflows/server-build.yml [hunk 2]
```diff
@@ -110,23 +110,6 @@ jobs:
...
-        if: steps.run_result.outputs.run_result != 'success'
-        id: vars
-        run: |
-          # Since this is an unreleased build, we set the version to incremented version number with a
-          # `-SNAPSHOT` suffix.
-          latest_released_version="$(git tag --list 'v*' --sort=-version:refname | head -1)"
-          echo "latest_released_version = $latest_released_version"
-          next_version="$(echo "$latest_released_version" | awk -F. -v OFS=. '{ $NF++; print }')"
-          echo "next_version = $next_version"
-          echo version=$next_version-SNAPSHOT >> $GITHUB_OUTPUT
```
Comment (sharat87) [inline .github/workflows/server-build.yml:127]: This variable is used in the next step. Are we sure we want to remove this?

File: .github/workflows/server-spotless.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# This workflow is responsible for running Spotless check on server code base
+name: Appsmith Server Spotless Check Workflow
```
Comment (sharat87) [inline .github/workflows/server-spotless.yml:2]: Minor.

```suggestion
name: Server Spotless Check
```

File: .github/workflows/formatting-checks.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+name: Perform formatting checks
```
Comment (sharat87) [inline .github/workflows/formatting-checks.yml:1]:
```suggestion
name: Formatting checks
```

File: .github/workflows/client-prettier.yml (hunk 1 above)
Comment (nidhi-nair) [reply .github/workflows/client-prettier.yml:1]: Agreed, felt awkward writing it as well :P

File: .github/workflows/server-build.yml (hunk 2 above)
Comment (nidhi-nair) [reply .github/workflows/server-build.yml:127]: Oh that's true, I only checked for tag. Thanks for catching this, I'll revert the entire block

File: .github/workflows/formatting-checks.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
+name: Formatting checks
+
+on:
+ pull_request:
```
Comment (sharat87) [inline .github/workflows/formatting-checks.yml:4]:
```suggestion
  pull_request:
```

**Review by sharat87 — APPROVED** [review]

**Pull request merged**